src/main/java/org/opentripplanner/routing/api/request/request/filter/SelectRequest.java

```diff
@@ -26,16 +25,16 @@ public static Builder of() {
   // TODO: 2022-11-29 filters: group of routes
 
   public SelectRequest(Builder builder) {
-    if (!builder.transportModes.isEmpty()) {
-      this.transportModeFilter = AllowTransitModeFilter.of(builder.transportModes);
-    } else {
+    if (builder.transportModes.isEmpty()) {
       this.transportModeFilter = null;
+    } else {
+      this.transportModeFilter = AllowTransitModeFilter.of(builder.transportModes);
     }
 
     // TODO: 2022-12-20 filters: having list of modes and modes filter in same instance is not very elegant
     this.transportModes = builder.transportModes;
 
-    this.agencies = Collections.unmodifiableList(builder.agencies);
+    this.agencies = List.copyOf(builder.agencies);
```

Comment:
Using `Collections.unmodifiableList(builder.agencies)` allows the list of agencies to be modified outside of this class - it break encapsulation.